charts/opentelemetry-collector/templates/_helpers.tpl

```diff
@@ -139,13 +139,6 @@ Return if ingress is stable.
 {{- end }}
 {{- end }}
 
-{{- define "opentelemetry-collector.annotations" -}}
-{{- if and (eq .Values.mode "deployment") .Values.annotations }}
```

Comment:
Instead of removing this helper, can we at statefulset to the condition?

Reply:
I actually agree @sherifkayad. I don't think why the helper is needed anymore. @TylerHelmuth don't we want to apply annotation to daemonset?

Reply:
I don't have a good reason other than at some point we decided to exclude them for daemonsets. I can't think of a good reason to do that so you're both right, removing this is a good idea.

Reply:
@dmitryax @TylerHelmuth thanks guys for the feedback. So I understand we can get this one merged already?